lib/sequent/core/helpers/param_support.rb

```diff
@@ -45,36 +51,54 @@ def as_params
           self.class.types.each do |field|
             value = self.instance_variable_get("@#{field[0]}")
             next if field[0] == "errors"
-            if value.respond_to?(:as_params) && value.kind_of?(ValueObject)
-              value = value.as_params
-            elsif value.kind_of?(Array)
-              value = value.map { |val| val.kind_of?(ValueObject) ? val.as_params : val }
-            elsif value.is_a? DateTime
-              value = value.iso8601
-            elsif value.is_a? Date
-              value = value.strftime("%d-%m-%Y") # TODO Remove to TypeConverter
-            end
-            hash[field[0]] = value
+            hash[field[0]] = if value.kind_of?(Array)
```

Comment:
I think we eventually should, this would I guess be another PR since we have to look into ArrayWithType and array(Integer) etc.